plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -572,35 +581,43 @@ int loadRuleFromCacheOrFile( const char* inst_name, const char *irbSet ) {
+                        int ret = hash_rules_with_copy(irbs, hash, stored_rulebases);
 
-                    in_memory_rulebases copy_rule_base_files(irbs);
-                    std::string hash;
-                    int ret = hash_rules_with_copy(irbs, hash, copy_rule_base_files);
-
-                    int diffHash = ret < 0 || hash != cache->hash;
-                    if ( diffIrbSet ) {
-                        rodsLog( LOG_DEBUG, "Rule base set changed, old value is %s", cache->ruleBase );
-                    }
+                        int diffHash = ret < 0 || hash != cache->hash;
+                        if (diffIrbSet) {
+                            rodsLog(LOG_DEBUG, "Rule base set changed, old value is %s", cache->ruleBase);
+                        }
 
-                    if ( diffIrbSet || time_type_gt( timestamp, cache->timestamp ) || diffHash ) {
-                        update = 1;
-                        free( cache->address );
-                        rodsLog( LOG_DEBUG, "Rule base set or rule files modified, force refresh." );
-                    } else {
-                        cache->cacheStatus = INITIALIZED;
-                        ruleEngineConfig = *cache;
-
-                        /* generate extRuleSet */
-                        generateRegions();
-                        generateRuleSets();
-                        generateFunctionDescriptionTables();
-                        if ( ruleEngineConfig.ruleEngineStatus == UNINITIALIZED ) {
-                            getSystemFunctions( ruleEngineConfig.sysFuncDescIndex->current, ruleEngineConfig.sysRegion );
+                        if (diffIrbSet || time_type_gt(timestamp, cache->timestamp) || diffHash) {
+                            update = 1;
+                            free(cache->address);
+                            rodsLog(LOG_DEBUG, "Rule base set or rule files modified, force refresh.");
                         }
+                        else {
+                            cache->cacheStatus = INITIALIZED;
+                            ruleEngineConfig = *cache;
+
+                            /* generate extRuleSet */
+                            generateRegions();
+                            generateRuleSets();
+                            generateFunctionDescriptionTables();
+                            if (ruleEngineConfig.ruleEngineStatus == UNINITIALIZED) {
+                                getSystemFunctions(
+                                    ruleEngineConfig.sysFuncDescIndex->current, ruleEngineConfig.sysRegion);
+                            }
 
-                        ruleEngineConfig.ruleEngineStatus = INITIALIZED;
+                            ruleEngineConfig.ruleEngineStatus = INITIALIZED;
 
-                        return res;
+                            return res;
+                        }
+                    }
+                    catch (const irods::exception& e) {
+                        logger::rule_engine::error(
```

Comment:
I see `log_re::error` used above?   Do we want / need consistency?

Reply:
Yes, please replace.